src/scratchOrgInfoApi.ts

```diff
@@ -0,0 +1,504 @@
+    hubOrg,
+    clientSecret,
+    scratchOrgInfoComplete,
+    retry: options?.retry,
+    signupTargetLoginUrlConfig,
+  });
+
+  await Lifecycle.getInstance().emit('authorizeScratchOrg', {
+    status: 'getAuthInfo',
+  });
+
+  const authInfo = await getAuthInfo({
+    hubOrg,
+    username: scratchOrgInfoComplete.SignupUsername,
+    oauth2Options: oAuth2Options.options,
+    retries: oAuth2Options.retries,
+    timeout: oAuth2Options.timeout,
+    delay: oAuth2Options.delay,
+  });
+
+  await saveAuthInfo({
+    scratchOrgInfoComplete,
+    hubOrg,
+    authInfo,
+    setAsDefault,
+    alias,
+  });
+
+  return authInfo;
+};
+
+const checkOrgDoesntExist = async (scratchOrgInfo: Record<string, unknown>): Promise<void> => {
+  const usernameKey = Object.keys(scratchOrgInfo).find((key: string) =>
+    key ? key.toUpperCase() === 'USERNAME' : false
+  );
+  if (!usernameKey) {
+    return;
+  }
+
+  const username = getString(scratchOrgInfo, usernameKey);
+
+  if (username && username.length > 0) {
+    try {
+      await AuthInfo.create({ username: username.toLowerCase() });
+    } catch (error) {
+      const sfdxError = SfdxError.wrap(error as Error);
+      // if an AuthInfo couldn't be created that means no AuthFile exists.
+      if (sfdxError.name === 'NamedOrgNotFound') {
+        return;
+      }
+      // Something unexpected
+      throw error;
+    }
+    // An org file already exists
+    throw SfdxError.create('scratchOrgInfoApi', 'signup', 'C-1007');
+  }
+};
+
+/**
+ * This extracts orgPrefs/settings from the user input and performs a basic scratchOrgInfo request.
+ *
+ * @param hubOrg - the environment hub org
+ * @param scratchOrgRequest - An object containing the fields of the ScratchOrgInfo
+ * @param settings - An object containing org settings
+ * @returns {Promise<RecordResult>}
+ */
+export const requestScratchOrgCreation = async (
+  hubOrg: Org,
+  scratchOrgRequest: ScratchOrgInfo,
+  settings: SettingsGenerator
+): Promise<RecordResult> => {
+  // If these were present, they were already used to initialize the scratchOrgSettingsGenerator.
+  // They shouldn't be submitted as part of the scratchOrgInfo.
+  delete scratchOrgRequest.settings;
+  delete scratchOrgRequest.objectSettings;
+
+  // We do not allow you to specify the old and the new way of doing post create settings
+  if (scratchOrgRequest.orgPreferences && settings.hasSettings()) {
+    // This is not allowed
+    throw new SfdxError('signupDuplicateSettingsSpecified');
+  }
+
+  // deprecated old style orgPreferences
+  if (scratchOrgRequest.orgPreferences) {
+    throw new SfdxError(messages.getMessage('deprecatedPrefFormat'));
+  }
+
+  const scratchOrgInfo = mapKeys(scratchOrgRequest, upperFirst, true);
+
+  await checkOrgDoesntExist(scratchOrgInfo); // throw if it does exists.
+  try {
+    return await hubOrg.getConnection().sobject('ScratchOrgInfo').create(scratchOrgInfo);
+  } catch (error) {
+    const jsForceError = error as JsForceError;
```

Comment:
Can this be a `SfdxError` - I don't think we should be using `JsForceError` here, or if we need to, can you add a comment explaining why